analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Mutex.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Threading;
+
+namespace Net6
+{
+    // https://docs.microsoft.com/en-us/dotnet/standard/threading/mutexes
+    internal class NetMutexTest
+    {
+        public static void AcquiredNotReleased(Mutex paramMutex)
+        {
+            // Note that dispose does not release the resource
+
+            using (var m = new Mutex(true, "foo")) // FN, 'true' means it owns the mutex if no exception gets thrown
+            {
+                // do stuff
+            }
+
+            var m0 = new Mutex(initiallyOwned: true, "bar", out var m0WasCreated); // FN
+            m0.Dispose();
```

Comment:
All these cases (except `m5`) are not FNs, because the lock is not released on at least one execution path.

RSPEC says:
>If a lock is acquired and released within a method, **then** it must be released along all execution paths of that method.